core/src/main/java/org/apache/iceberg/ManifestFiles.java

```diff
@@ -55,6 +55,8 @@ public static ManifestReader read(ManifestFile manifest, FileIO io) {
    * @return a {@link ManifestReader}
    */
   public static ManifestReader read(ManifestFile manifest, FileIO io, Map<Integer, PartitionSpec> specsById) {
+    Preconditions.checkArgument(manifest.content() == ManifestContent.DATA,
```

Comment:
For my understanding,  the DATA manifest & DELETE manifest could share the same read / write path so I think we could use the common reader+writer.  Is there any other reason that we need to make them separate paths ?

Reply:
They have the same schema to keep the format simpler, but in the APIs we want to keep them separate by using DataFile and DeleteFile types. The readers and writers mostly share the same code, but are separate so that they can use the correct file type interface.